src/graph/planner/plan/PlanNode.cpp

```diff
@@ -379,6 +379,22 @@ void PlanNode::updateSymbols() {
   }
 }
 
+Status PlanNode::pruneProperties(PropertyTracker& propsUsed,
```

Comment:
I think we could do this by using the visitor pattern.

Reply:
Yes, you are right. Putting the method `pruneProperties` to the Plan node is a little intrusive. I will use the visitor pattern.